code-reviews.md

```diff
@@ -0,0 +1,141 @@
+- Push commits based on earlier rounds of feedback as isolated commits to the
+  branch. Do not squash until the branch is ready to merge. Reviewers should be
+  able to read individual updates based on their earlier feedback.
+- Wait to merge the branch until it has passed Continuous Integration testing.
+  (TDDium, TravisCI, etc.)
+- Merge once you feel confident in the code and its impact on the project.
+
+### For code reviewers
+
+- Understand why the code is necessary (bug, user experience, refactoring)
+- Seek to understand the author's perspective.
+- Clearly communicate which ideas you feel strongly about and those you don't.
+- Identify ways to simplify the code while still solving the problem.
+- Offer alternative implementations, but assume the author already considered
+  them. ("What do you think about such-and-such here?")
+- If discussions turn too philosophical or academic, move the discussion to a new
+  issue or offline. In the meantime, the author has the final say on the current
+  implementation.
+- Sign off on the pull request with a :thumbsup: or "Ready to merge" comment.
+
+
+## Submitting code
+
+Before seeking a review, you should be able to check off each of the following:
+
+[ ] Changes are limited to a single goal (no scope creep)
+[ ] Code can be automatically merged (no conflicts)
+[ ] Code follows the standards laid out in this playbook
+[ ] Passes all existing automated tests
+[ ] New functions include new tests
+[ ] New functions are documented (with a description, list of inputs, and
+    expected output)
+[ ] Placeholder code is flagged
+[ ] Visually tested in supported browsers and devices
+[ ] Project documentation has been updated (including the "Pending" section of
+    the CHANGELOG)
+
+### Opening a PR
+
+Once sure your code meets the checks above, open a Pull Request for your new
+feature or bugfix following our [PR Template](https://raw.githubusercontent.com/cfpb/front-end/733397192b04551d56c847f3416b0bcdf5bab544/pr-template.md). Remember that the
+more information you provide to reviewers, the more context they will have. This
+leads to faster reviews, and less back and forth between everyone.
+
+### Ready to merge?
+
+You've got your sign-offs, now what? If you've made any review edits, it's best
+to squash them to keep the history clean (though not required). Use the merge button
+on Github (don't merge locally) and ensure any automated deployments succeed
+(typically found in the project's chat channel). If the PR included a versioned
+changed to the CHANGELOG, be sure to create and push a matching tag.
+
+
+## Reviewing code
+
+When reviewing code, you should be able to check off each of the following:
+
+[ ] Do the changes address the project's needs?
+[ ] Do the changes respect the project's existing style?
+[ ] Does the new code avoid reproducing existing functionality?
+[ ] Are functions as simple as possible?
+[ ] Is the code as efficient as possible?
```

Comment:
Not exactly. Simplicity is about the code, how much does a function do? Verses efficiency is about the processing, how much work does it create.

One example would be:

```
// This is simpler
for ( i = 0; i < thing.length; i++ ) {
  //do things
}

// This is more efficient because it doesn't have to lookup the length on every loop
var i = 0;
var l = thing.length;
for ( ; i < l; i++ ) {
  //do things
}
```